### PR TITLE
Create and implement all in maze-gen

### PR DESCRIPTION
**Pull request by dapeef**
Created a maze generator, just the basics without removing extra random walls

### CODE REVIEW

File: maze-gen.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import random
+
+
+def generate_maze(size_x, size_y, density=0.9, print_map=False):
+    nodes = [Node(i + j * size_x, (i, j)) for j in range(size_y)
+             for i in range(size_x)]
+
+    walls = [
+        Wall("r", (i, j), True) for j in range(size_y)
+        for i in range(size_x - 1)
```
Comment (kneasle) [inline maze-gen.py:12]: Please could make these `for` loops line up.

File: maze-gen.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    bottom_walls = [[True for i in range(size_x)] for i in range(size_y - 1)]
+
+    for wall in walls:
+        if wall.orientation == "r":
+            right_walls[wall.position[1]][wall.position[0]] = wall.enabled
+
+        if wall.orientation == "b":
+            bottom_walls[wall.position[1]][wall.position[0]] = wall.enabled
+
+    if print_map:
```
Comment (kneasle) [inline maze-gen.py]: Please move this code into its own function, and make it private (by adding an underscore to the start).

**Review by kneasle — CHANGES_REQUESTED** [review]

**Comment by dapeef** [comment]
All requests resolved :)

**Comment by dapeef** [comment]
Take 2

**Pull request merged**